BuildScript.Nuke.UnitTests/SemanticVersionTest.cs

```diff
@@ -109,6 +111,71 @@ public void DependDBProjectVersion_WithValidVersion_ReturnsDepenDBProjectVersion
   [TestCase("wrong")]
   public void Version_WithInvalidVersion_ThrowsNukeBuildException (string version)
   {
-    Assert.That(() => new SemanticVersion(version), Throws.InstanceOf<ApplicationException>());
+    Assert.That(() => new SemanticVersion(version, false), Throws.InstanceOf<ApplicationException>());
+  }
+
+  [Test]
+  [TestCase("3.0.0")]
+  [TestCase("3.1.1")]
+  [TestCase("3.0.0-alpha.30")]
+  [TestCase("1.21.8-beta.102")]
+  public void Version_WithLocalBuildTrue_ReturnsValidVersionWithLocalBuildSuffix (string version)
+  {
+    string[] format = { "yyMMdd-HHmmss" };
+    DateTime parseDateTime;
+
+    var semanticVersion = new SemanticVersion(version, true);
+    var actualVersion = semanticVersion.Version;
+
+    var actualVersionDateTime = actualVersion.Split('.').Last();
+    var isValidDaTime = DateTime.TryParseExact(actualVersionDateTime, format, new CultureInfo("en-US"), DateTimeStyles.None, out parseDateTime);
+
+    Assert.That(actualVersion, Does.Contain("-x.9."));
+    Assert.That(isValidDaTime, Is.True);
+  }
+
+  [Test]
+  [TestCase("3.0.0", "Debug", "")]
+  [TestCase("3.1.0", "Release", "")]
+  [TestCase("3.0.0-alpha.30", "Release", "")]
+  [TestCase("1.21.8-beta.102", "Release", "buildMetaData")]
+  [TestCase("1.21.8-beta.102", "Release", "buildMetaData3422")]
+  public void GetAssemblyInformationVersion_WithLocalBuildTrue_ReturnsAssemblyInformationVersionWithLocalBuildSuffix (
+      string version,
+      string configurationId,
+      string additionalBuildMetadata)
+  {
+    string[] format = { "yyMMdd-HHmmss" };
+    DateTime parseDateTime;
+
+    var semanticVersion = new SemanticVersion(version, true);
+    var assemblyInformationVersion = semanticVersion.GetAssemblyInformationalVersion(configurationId, additionalBuildMetadata);
+
+    var splittedVersion = assemblyInformationVersion.Split('.');
+    var actualVersionDateTime = splittedVersion.Single(x => x.Contains('+')).Split('+').First();
+    var isValidDaTime = DateTime.TryParseExact(actualVersionDateTime, format, new CultureInfo("en-US"), DateTimeStyles.None, out parseDateTime);
+
+    Assert.That(assemblyInformationVersion, Does.Contain("-x.9."));
```

Comment:
I think it would be better to use regex expressions to assert the version structure, what do you think?